### PR TITLE
Set preset in Spring Config

### PR DESCRIPTION
**Pull request by pascalgrimaud**
- fix https://github.com/jhipster/jhipster-lite/issues/10553

### CODE REVIEW

File: src/main/java/tech/jhipster/lite/project/infrastructure/secondary/FileSystemProjectsRepository.java [hunk 1]
```diff
@@ -26,18 +27,24 @@ class FileSystemProjectsRepository implements ProjectsRepository {
...
-  private static final String PRESET_FILE = "preset.json";
 
   private final ObjectMapper json;
   private final ProjectFormatter formatter;
   private final ProjectFiles projectFiles;
   private final ObjectWriter writer;
   private final FileSystemProjectDownloader downloader;
-
-  public FileSystemProjectsRepository(ObjectMapper json, ProjectFormatter formatter, ProjectFiles projectFiles) {
+  private final String presetFile;
```
Comment (renanfranca) [inline src/main/java/tech/jhipster/lite/project/infrastructure/secondary/FileSystemProjectsRepository.java:36]: I think it should use

```suggestion
  private final PresetName presetName;
```

File: src/main/java/tech/jhipster/lite/project/infrastructure/secondary/FileSystemProjectsRepository.java [hunk 2]
```diff
@@ -26,18 +27,24 @@ class FileSystemProjectsRepository implements ProjectsRepository {
...
+
+  public FileSystemProjectsRepository(
+    ObjectMapper json,
+    ProjectFormatter formatter,
+    ProjectFiles projectFiles,
+    @Value("${jhlite-preset-file.name:preset.json}") String presetFile
```
Comment (renanfranca) [inline src/main/java/tech/jhipster/lite/project/infrastructure/secondary/FileSystemProjectsRepository.java:42]:
```suggestion
    @Value("${jhlite-preset-file.name:preset.json}") String presetFileName
```

File: src/main/java/tech/jhipster/lite/project/infrastructure/secondary/FileSystemProjectsRepository.java [hunk 3]
```diff
@@ -102,6 +109,6 @@ public Collection<Preset> getPresets() {
   }
 
   private String presetFilePath() {
-    return PRESET_FOLDER + PRESET_FILE;
+    return PRESET_FOLDER + presetFile;
```
Comment (renanfranca) [inline src/main/java/tech/jhipster/lite/project/infrastructure/secondary/FileSystemProjectsRepository.java:112]:
```suggestion
    return PRESET_FOLDER + presetName.name();
```

**Comment by renanfranca** [comment]
@pascalgrimaud : A error was detected with `custom-jhlite` module, what do you think about this fix?
https://github.com/jhipster/jhipster-lite/pull/10609#issuecomment-2296549821

This must be done in this same PR.

**Comment by pascalgrimaud** [comment]
@renanfranca : I tested locally jhlite custom with the following code here and it works, as I used default override mechanism by Spring, so the property will be set with the default value preset.json

**Comment by renanfranca** [comment]
> @renanfranca : I tested locally jhlite custom with the following code here and it works, as I used default override mechanism by Spring, so the property will be set with the default value preset.json

Nice! Thanks!

@pascalgrimaud : If you change the property at `application.yml` the cucumber test will fail. To avoid that the class `FakedFileSystemProjectFilesConfiguration.java` should be change:

### the line to change:
from:
`Mockito.doReturn(presetJsonContent.getBytes(StandardCharsets.UTF_8)).when(fileSystemProjectFiles).readBytes("/preset.json");`

to:
`    Mockito.doReturn(presetJsonContent.getBytes(StandardCharsets.UTF_8)).when(fileSystemProjectFiles).readBytes(anyString());`


### The class without and with the change:
from:
```java
ackage tech.jhipster.lite.project.infrastructure.secondary;

import java.nio.charset.StandardCharsets;
import org.mockito.Mockito;
import org.springframework.boot.test.context.TestConfiguration;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Primary;
import tech.jhipster.lite.module.infrastructure.secondary.FileSystemProjectFiles;

@TestConfiguration
public class FakedFileSystemProjectFilesConfiguration {

  @Bean
  @Primary
  FileSystemProjectFiles fileSystemProjectFiles() {
    FileSystemProjectFiles fileSystemProjectFiles = Mockito.spy(new FileSystemProjectFiles());

    mockPresetJson(fileSystemProjectFiles);

    return fileSystemProjectFiles;
  }

  private static void mockPresetJson(FileSystemProjectFiles fileSystemProjectFiles) {
    String presetJsonContent =
      """
      {
        "presets": [
          {
            "name": "test preset one",
            "modules": ["test-module-one", "test-module-two"]
          },
          {
            "name": "test preset two",
            "modules": ["test-module-three", "test-module-four"]
          }
        ]
      }
      """;

    Mockito.doReturn(presetJsonContent.getBytes(StandardCharsets.UTF_8)).when(fileSystemProjectFiles).readBytes("/preset.json");
  }
}
```

to:
```java
package tech.jhipster.lite.project.infrastructure.secondary;

import static org.mockito.ArgumentMatchers.*;

import java.nio.charset.StandardCharsets;
import org.mockito.Mockito;
import org.springframework.boot.test.context.TestConfiguration;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Primary;
import tech.jhipster.lite.module.infrastructure.secondary.FileSystemProjectFiles;

@TestConfiguration
public class FakedFileSystemProjectFilesConfiguration {

  @Bean
  @Primary
  FileSystemProjectFiles fileSystemProjectFiles() {
    FileSystemProjectFiles fileSystemProjectFiles = Mockito.spy(new FileSystemProjectFiles());

    mockPresetJson(fileSystemProjectFiles);

    return fileSystemProjectFiles;
  }

  private static void mockPresetJson(FileSystemProjectFiles fileSystemProjectFiles) {
    String presetJsonContent =
      """
      {
        "presets": [
          {
            "name": "test preset one",
            "modules": ["test-module-one", "test-module-two"]
          },
          {
            "name": "test preset two",
            "modules": ["test-module-three", "test-module-four"]
          }
        ]
      }
      """;

    Mockito.doReturn(presetJsonContent.getBytes(StandardCharsets.UTF_8)).when(fileSystemProjectFiles).readBytes(anyString());
  }
}

```

**Comment by pascalgrimaud** [comment]
@renanfranca : nice catch and thanks. See my last commit, we should not need to apply this additional code about Cucumber

**Comment by renanfranca** [comment]
> > @renanfranca : I tested locally jhlite custom with the following code here and it works, as I used default override mechanism by Spring, so the property will be set with the default value preset.json
> 
> Nice! Thanks!
> 
> @pascalgrimaud : If you change the property at `application.yml` the cucumber test will fail. To avoid that the class `FakedFileSystemProjectFilesConfiguration.java` should be change:
> 
> ### the line to change:
> from: `Mockito.doReturn(presetJsonContent.getBytes(StandardCharsets.UTF_8)).when(fileSystemProjectFiles).readBytes("/preset.json");`
> 
> to: ` Mockito.doReturn(presetJsonContent.getBytes(StandardCharsets.UTF_8)).when(fileSystemProjectFiles).readBytes(anyString());`
> 
> ### The class without and with the change:
> from:
> 
> ```java
> ackage tech.jhipster.lite.project.infrastructure.secondary;
> 
> import java.nio.charset.StandardCharsets;
> import org.mockito.Mockito;
> import org.springframework.boot.test.context.TestConfiguration;
> import org.springframework.context.annotation.Bean;
> import org.springframework.context.annotation.Primary;
> import tech.jhipster.lite.module.infrastructure.secondary.FileSystemProjectFiles;
> 
> @TestConfiguration
> public class FakedFileSystemProjectFilesConfiguration {
> 
>   @Bean
>   @Primary
>   FileSystemProjectFiles fileSystemProjectFiles() {
>     FileSystemProjectFiles fileSystemProjectFiles = Mockito.spy(new FileSystemProjectFiles());
> 
>     mockPresetJson(fileSystemProjectFiles);
> 
>     return fileSystemProjectFiles;
>   }
> 
>   private static void mockPresetJson(FileSystemProjectFiles fileSystemProjectFiles) {
>     String presetJsonContent =
>       """
>       {
>         "presets": [
>           {
>             "name": "test preset one",
>             "modules": ["test-module-one", "test-module-two"]
>           },
>           {
>             "name": "test preset two",
>             "modules": ["test-module-three", "test-module-four"]
>           }
>         ]
>       }
>       """;
> 
>     Mockito.doReturn(presetJsonContent.getBytes(StandardCharsets.UTF_8)).when(fileSystemProjectFiles).readBytes("/preset.json");
>   }
> }
> ```
> 
> to:
> 
> ```java
> package tech.jhipster.lite.project.infrastructure.secondary;
> 
> import static org.mockito.ArgumentMatchers.*;
> 
> import java.nio.charset.StandardCharsets;
> import org.mockito.Mockito;
> import org.springframework.boot.test.context.TestConfiguration;
> import org.springframework.context.annotation.Bean;
> import org.springframework.context.annotation.Primary;
> import tech.jhipster.lite.module.infrastructure.secondary.FileSystemProjectFiles;
> 
> @TestConfiguration
> public class FakedFileSystemProjectFilesConfiguration {
> 
>   @Bean
>   @Primary
>   FileSystemProjectFiles fileSystemProjectFiles() {
>     FileSystemProjectFiles fileSystemProjectFiles = Mockito.spy(new FileSystemProjectFiles());
> 
>     mockPresetJson(fileSystemProjectFiles);
> 
>     return fileSystemProjectFiles;
>   }
> 
>   private static void mockPresetJson(FileSystemProjectFiles fileSystemProjectFiles) {
>     String presetJsonContent =
>       """
>       {
>         "presets": [
>           {
>             "name": "test preset one",
>             "modules": ["test-module-one", "test-module-two"]
>           },
>           {
>             "name": "test preset two",
>             "modules": ["test-module-three", "test-module-four"]
>           }
>         ]
>       }
>       """;
> 
>     Mockito.doReturn(presetJsonContent.getBytes(StandardCharsets.UTF_8)).when(fileSystemProjectFiles).readBytes(anyString());
>   }
> }
> ```

@pascalgrimaud 

> If you change the property at `application.yml` the cucumber test will fail. To avoid that the class `FakedFileSystemProjectFilesConfiguration.java` should be change:
> 
> ### the line to change:
> from: `Mockito.doReturn(presetJsonContent.getBytes(StandardCharsets.UTF_8)).when(fileSystemProjectFiles).readBytes("/preset.json");`
> 
> to: ` Mockito.doReturn(presetJsonContent.getBytes(StandardCharsets.UTF_8)).when(fileSystemProjectFiles).readBytes(anyString());`

@pascalgrimaud : please, hold on. Don't implement that change, it will break a lot of other tests and won't fix the problem when the `preset-filename.name` is changed!

I don't have a solution right now from the top of my head.

**Comment by renanfranca** [comment]
> @renanfranca : nice catch and thanks. See my last commit, we should not need to apply this additional code about Cucumber

@pascalgrimaud : Greate! Thanks for the solution, I didn't see that option implement the code in my head 😅

**Review by renanfranca — APPROVED** [review]
Thanks @pascalgrimaud for your work! 😃

**Pull request merged**